lib/index.d.ts

```diff
@@ -0,0 +1,214 @@
+import webpack = require("webpack");
+
+type WebpackUserscriptOptions =
+  WPUSOptions |
+  HeaderFile |    // shorthand for WPUSOptions.headers
+  HeaderProvider  // shorthand for WPUSOptions.headers
+
+interface WPUSOptions {
+  headers: HeaderFile | HeaderProvider | HeaderObject
```

Comment:
`headers` is also optional, some required header will be parsed from `package.json`.